soroban-env-host/src/vm.rs

```diff
@@ -347,7 +347,14 @@ impl Vm {
             match e {
                 wasmi::Error::Trap(trap) => {
                     if let Some(code) = trap.trap_code() {
-                        return Err(code.into());
+                        let err = code.into();
+                        return Err(if host.is_debug()? {
+                            // With diagnostics on: log as much detail as we can from wasmi.
+                            let msg = format!("VM call trapped: {:?}", &code);
+                            host.error(err, &msg, &[func_sym.to_val()])
```

Comment:
```suggestion
                            host.error(err, "VM call trapped", &[func_sym.to_val(), code.to_val()])
```

Reply:
Should be `err_to_val()` instead, updated.